### DISCUSSION TITLE
i can not to send formdata to request.file Adonis 5, request.file is null

### DISCUSSION THREAD

**Discussion by Scute300** [adonisjs/core#1920] [Help]
sorry for the question, in this moment i have two days finding and i do not found some fix for my problem. i making a aplication using expo-react native at the frontend and adonis 5 at the backend, im triying to send a image from expo to adonis and save image in public directory the problem is that adonis does not receive the file, always receive null. I am sure that this is for a problem on my way of to send form data but i can not found the problem...

This is my current code at the frontend:

`
//Component 

    const imagepreview = async()=>{
        let result = await ImagePicker.launchImageLibraryAsync({
            allowsEditing: true,
            aspect: [4, 3],
            quality: 1,
        });
        console.log(result)
        if(result.cancelled == false){
            setImage(result.uri)
        }
    }

    const sendData = async()=>{
      
        // Upload the image using the fetch and FormData APIs
        let formData = new FormData();
        // Assume "photo" is the name of the form field the server expects
        formData.append('photo', {type:'image/jpg', uri:image, name:'lorem'});
        const send = await user.changeprofilepicture(formData)

    }
    
 //Class for request

    async changeprofilepicture(formdata){
        console.log(formdata)
        const token = JSON.parse(localStorage.getItem('user-token'))
        let url = baseurl+'changeprofilepicture'
        let config = {
          headers: {
            'content-type': 'multipart/form-data'
          }
        }

        await axios({
            method: 'post',
            url: url,
            data:  formdata,
            config: config
       })      
      
    }`

And my adonis Code : 

//code
    
```
async changeprofilepicture({request, response}:HttpContextContract){
        try{
            const file = request.file('photo')
            console.log(file)
            await file!.move(Application.tmpPath('assets'), {
                name: 'tryes',
                overwrite: true
              })
            return response.json({
                status: 'sure'
            })
    

        }catch (e){
            console.log(e)
        }
    }

```
Console.log is null but when i try to use "request.all()" Method adonis showme this : { photo: '[object Object]' } but i can not move... ¿Any idea? Thanks for the answers!

Update:  I am try to take another way : 
`
        formData.append('photo', { uri: localUri, name: filename, type });
      
        return await fetch('http://127.0.0.1:3333/api/changeprofilepicture', {
          method: 'POST',
          body: formData,
          headers: {
            'content-type': 'multipart/form-data',
          },
        });`

But i received this error message : 

"message":"content-type missing boundary","stack":"BadRequestError: content-type missing boundary\n    at Form.parse (/home/antonio/Documentos/naviera/romaBackend/node_modules/multiparty/index.js:181:21)\n    at /home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/bodyparser/build/src/Multipart/index.js:281:23\n    at new Promise (<anonymous>)\n    at Multipart.process (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/bodyparser/build/src/Multipart/index.js:222:16)\n    at BodyParserMiddleware.handle (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/bodyparser/build/src/BodyParser/index.js:146:45)\n    at Injector.injectMethodDependencies (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/fold/build/src/Ioc/Injector.js:67:30)\n    at Ioc.call (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/fold/build/src/Ioc/index.js:488:30)\n    at IocResolver.call (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/fold/build/src/Resolver/index.js:100:31)\n    at MiddlewareStore.invokeMiddleware (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/http-server/build/src/MiddlewareStore/index.js:104:30)\n    at Runnable.RequestHandler.executeMiddleware [as executorFn] (/home/antonio/Documentos/naviera/romaBackend/node_modules/@adonisjs/http-server/build/src/Server/RequestHandler/index.js:29:41)"}

¿Any idea?

**Comment by FeMaffezzolli** [comment]
Why do you have an exclamation mark after your `file` variable in `file!.move`?
